autoclasstoc/sections.py

```diff
@@ -114,6 +123,16 @@ def format(self):
 
         return [wrapper]
 
+    def check_for_pattern(self, name):
+        """
+        Return true if the given :attr:`name` matches the 
+        :attr:`~.exclude_pattern`
+        """
+        if self.exclude_pattern:
+            return any(
+                re.search(p, name)
```

Comment:
The question here is `match` or `search`; the latter seems to _search_ the whole `name` while the first one "just" looks from the beginning.